onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -2478,7 +2446,33 @@ namespace OperatorHelper
             if (kernelInformation.IsInputValid(3))
             {
                 MLOperatorTensor outputSizesTensor = kernelInformation.GetConstantInputTensor(3);
-                ReadCpuLocalTensorIntoInt32(outputSizesTensor, /*out*/ outputSizes);
+                ReadCpuLocalTensorIntoInt32<uint32_t>(outputSizesTensor, /*out*/ outputSizes);
+            }
+
+            axes = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Axes);
+            // Handle possible axes input
+            if (opsetVersion >= 18 && !axes.empty())
+            {
+                uint32_t dimCount = gsl::narrow_cast<uint32_t>(m_inputDimensions.size());
+                HandleEmptyAxes(/*inout*/ axes, m_inputDimensions, false);
+                HandleNegativeAxes(/*inout*/ axes, dimCount);
+
+                // Taken from https://github.com/onnx/onnx/blob/3d69db8fd16873d68e7033479467f9478562a12d/onnx/reference/ops/op_resize.py#L303
+                if (!m_scales.empty())
+                {
+                    std::vector<float> newScales(dimCount, 1.0f);
+                    ExpandToAxes(/*inuout*/ m_scales, axes, newScales);
+                }
+                if (!outputSizes.empty())
+                {
+                    ExpandToAxes(/*inuout*/ outputSizes, axes, m_inputDimensions);
+                }
+                if (!m_regionOfInterest.empty())
+                {
+                    std::vector<float> newRois(dimCount, 0.0f);
+                    newRois.resize(dimCount * 2, 1.0f);
+                    ExpandToAxes(/*inuout*/ m_regionOfInterest, axes, newRois);
```

Comment:
Since ROI's consist of two concatenated arrays `[start1, …, startN, end1, …, endN]`, does this work as-is? It totally would work if ONNX stored them as pairs `[start1, end1,, …, startN, endN]`, and we used a `std::vector<std::pair<float, float>>`, but it looks like this would scramble the values. So I don't think we can use our nice little helper for ROI.